src/main/java/core/basesyntax/ReadWriteList.java

```diff
@@ -10,16 +10,28 @@
     private ReadWriteLock lock = new ReentrantReadWriteLock();
 
     public void add(E element) {
-        // write your code here
+        lock.writeLock().lock();
+        lock.readLock().lock();
```

Comment:
You don't need to lock the read lock when performing a write operation. Remove the line `lock.readLock().lock();`. 🧐